addons/mrp_subcontracting/models/stock_warehouse.py

```diff
@@ -28,19 +27,38 @@ class StockWarehouse(models.Model):
     def create(self, vals_list):
         res = super().create(vals_list)
         res._update_subcontracting_locations_rules()
+        # if new warehouse has resupply enabled, enable global route
+        if any(['subcontracting_to_resupply' in vals for vals in vals_list]):
+            res._update_global_route_resupply_subcontractor()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        # if all warehouses have resupply disabled, disable global route, until its enabled on a warehouse
+        if 'subcontracting_to_resupply' in vals or 'active' in vals:
```

Comment:
Also the `''active'` case is already handled here: https://github.com/odoo/odoo/blob/1b6af2addbe10c60b595846050faf82b9a23ed92/addons/stock/models/stock_warehouse.py#L234-L237
Your current change makes it unarchive the rule after it's been archived.

Reply:
that condition is for the function `_update_global_route_resupply_subcontractor` to run, since I need to check after the rules are archived in the `write` function if I need to archive global routes.

Reply:
Then it sounds like you need to put the `if 'subcontracting_to_resupply' in vals` and `self._update_resupply_rules()` check before the `res = super().write(vals)` and keep the `active` and `self._update_global_route_resupply_subcontractor()` after

Reply:
Sorry I was wrong. What you need to do is:

```
if 'subcontracting_to_resupply' in vals or 'active' in vals:
    if and not vals.get('active', False):
        # ignore when warehouse archived since it will auto-archive all of its rules
        self._update_resupply_rules()
    self._update_global_route_resupply_subcontractor()
```

and you can delete the part before the `res = super().write(vals)`